profiling/src/lib.rs

```diff
@@ -386,10 +400,10 @@ extern "C" fn rinit(r#type: c_int, module_number: c_int) -> ZendResult {
     ) = unsafe {
         (
             config::profiling_enabled(),
-            config::profiling_endpoint_collection_enabled(),
-            config::profiling_experimental_cpu_time_enabled(),
-            config::profiling_allocation_enabled(),
-            config::profiling_experimental_timeline_enabled(),
+            config::profiling_endpoint_collection_enabled() && config::profiling_enabled(),
+            config::profiling_experimental_cpu_time_enabled() && config::profiling_enabled(),
+            config::profiling_allocation_enabled() && config::profiling_enabled(),
+            config::profiling_experimental_timeline_enabled() && config::profiling_enabled(),
```

Comment:
Could you have a look at https://github.com/DataDog/dd-trace-php/pull/2189/commits/516a4bd8895e7c26939cd62c7f8bbb5eeaf277d9
This will only call the `config::profiling_...()` functions if `profiling_enabled` is true.

Reply:
I think what the commit has is good enough. I would _hope_ the optimizer can handle it anyway but even if it can't, it's better than before the commit.